Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -1759,6 +1759,42 @@ class BrowserViewController: UIViewController {
         duckDuckGoPopup = popup
         popup.showWithType(showType: .flyUp)
     }
+    
+    private func showBookmarkController() {
+        let bookmarkViewController = BookmarksViewController(
+            folder: nil,
+            isPrivateBrowsing: PrivateBrowsingManager.shared.isPrivateBrowsing)
+        
+        bookmarkViewController.toolbarUrlActionsDelegate = self
+        
+        presentSettingsNavigation(with: bookmarkViewController)
+    }
+    
+    func openAddBookmark() {
+        guard let selectedTab = tabManager.selectedTab, let selectedUrl = selectedTab.url else { return }
```

Comment:
we should add additional safeguard here to disable this hotkey action if we are on new tab page or while in reader mode, similar to what we do in `MenuViewController`

<img width="584" alt="Zrzut ekranu 2020-11-10 o 20 20 38" src="https://user-images.githubusercontent.com/6950387/98723185-56990b80-2392-11eb-9523-560af9f487cb.png">

Reply:
Safeguard is added, do you think we should add a safeguard for adding a bookmark from settings menu while the tab is in readermode ?

Reply:
You mean like we are in settings panel, then pressing CMD+D opens add-bookmark screen? If that's easy to do we could try to do that

Reply:
I was thinking about disabling clicking the addbookmark from the menu but I guess we can talk about it